sklearn/cluster/tests/test_k_medoids.py

```diff
@@ -0,0 +1,138 @@
+                         KMedoids(n_clusters=None).fit, X)
+
+    assert_raise_message(ValueError, "init needs to be one of the following: "
+                                     "['random', 'heuristic']",
+                         KMedoids(init=None).fit, X)
+
+    # Trying to fit 3 samples to 8 clusters
+    model = KMedoids(n_clusters=8)
+
+    Xsmall = rng.rand(5, 2)
+
+    assert_raises(ValueError, model.fit, Xsmall)
+
+    # Trying to transform without fiting
+    model = KMedoids()
+
+    assert_raises(NotFittedError, model.transform, X)
+
+    # Trygin to predict without fitting
+    model = KMedoids()
+
+    assert_raises(NotFittedError, model.predict, X)
+
+
+def test_kmedoids_fit_naive_with_all_pairwise_distance_functions():
+    for distance_metric in PAIRWISE_DISTANCE_FUNCTIONS.keys():
+
+        if distance_metric == 'precomputed':
+            continue
+
+        model = KMedoids(n_clusters=3, distance_metric=distance_metric)
+        Xnaive = np.asarray([[1, 0, 0], [0, 1, 0], [0, 0, 1]])
+
+        model.fit(Xnaive)
+
+        assert_array_equal(model.cluster_centers_,
+                           [[1, 0, 0], [0, 1, 0], [0, 0, 1]])
+        assert_array_equal(model.labels_, [0, 1, 2])
+        assert_equal(model.inertia_, 0.0)
+
+
+def test_kmedoids_iris_with_all_pairwise_distance_functions():
+    Xiris = load_iris()['data']
+
+    ref_model = KMeans(n_clusters=3)
+
+    ref_model.fit(Xiris)
+
+    avg_dist_to_closest_centroid = np.sum(np.min(
+        euclidean_distances(Xiris, Y=ref_model.cluster_centers_), axis=1)
+    ) / Xiris.shape[0]
+
+    for init in ['random', 'heuristic']:
+        for distance_metric in PAIRWISE_DISTANCE_FUNCTIONS.keys():
+            if distance_metric == 'precomputed':
+                continue
+
+            model = KMedoids(n_clusters=3,
+                             distance_metric=distance_metric,
+                             init=init,
+                             random_state=rng)
+            model.fit(Xiris)
+
+            distances = PAIRWISE_DISTANCE_FUNCTIONS[distance_metric](Xiris)
+            avg_dist_to_random_medoid = np.mean(distances.ravel())
+            avg_dist_to_closest_medoid = model.inertia_ / Xiris.shape[0]
+            # We want distance-to-closest-medoid to be reduced from average
+            # distance by more than 50%
+            assert_greater(0.5 * avg_dist_to_random_medoid,
+                           avg_dist_to_closest_medoid)
+            # When K-Medoids is using Euclidean distance,
+            # we can compare its performance to
+            # K-Means. We want to average distance to cluster centers
+            # be similar between K-Means and K-Medoids
+            if distance_metric == "euclidean":
+                assert_greater(0.1,
+                               np.abs(avg_dist_to_closest_medoid -
+                                      avg_dist_to_closest_centroid))
+
+
+def test_kmedoids_fit_predict():
```

Comment:
Could you combine both these tests into one? `test_kmedoids_fit_predict_transform`?